Student/StudentE2ETests/Dashboard/DashboardTests.swift

```diff
@@ -254,4 +254,70 @@ class DashboardTests: E2ETestCase {
         XCTAssertTrue(courseCard.isVisible)
         XCTAssertTrue(courseCardGradeLabel.isVanished)
     }
+
+    func testCourseCardReorder() {
+        // MARK: Seed the usual stuff with 2 courses
+        let student = seeder.createUser()
+        let courses = seeder.createCourses(count: 2)
+        seeder.enrollStudent(student, in: courses[0])
+        seeder.enrollStudent(student, in: courses[1])
+
+        // MARK: Get the user logged in, check course cards
+        logInDSUser(student)
+        let courseCard1 = Helper.courseCard(course: courses[0]).waitUntil(.visible)
+        let courseCard2 = Helper.courseCard(course: courses[1]).waitUntil(.visible)
+        XCTAssertTrue(courseCard1.isVisible)
+        XCTAssertTrue(courseCard2.isVisible)
+
+        // MARK: Reorder course cards, check if successful
+        let courseCard1FrameBefore = courseCard1.frame
+        let courseCard2FrameBefore = courseCard2.frame
+        courseCard1.tapAndHoldAndDragToElement(element: courseCard2)
+        let courseCard1FrameAfter = courseCard1.frame
+        let courseCard2FrameAfter = courseCard2.frame
+        XCTAssertEqual(courseCard1FrameAfter, courseCard2FrameBefore)
+        XCTAssertEqual(courseCard2FrameAfter, courseCard1FrameBefore)
```

Comment:
We could do a login/logout to check if the new order gets restored from the API or not because if the API update fails here the app will swallow it but will update its local storage.

Reply:
Done.